pkg/ddevapp/utils.go

```diff
@@ -522,6 +522,27 @@ func (app *DdevApp) CanUseHTTPOnly() bool {
 	return false
 }
 
+// CreateWebWorkingDir creates a web working_dir if it does not exist on the host side.
+// If the app has a custom working_dir for the web service, like `/var/www/html/some/deep/dir`,
+// where `/var/www/html/` path corresponds to the project root,
+// and `some/deep/dir` does not exist on the host side, we should create it.
+// Otherwise, Docker will create it as root user (when Mutagen is disabled).
+// This problem (particularly for Docker volumes) is described in
+// https://github.com/moby/moby/issues/2259
+func (app *DdevApp) CreateWebWorkingDir() {
+	if app.WorkingDir != nil {
```

Comment:
Do we need to do any of this logic, or just use the working dir whatever it is, and os.MkdirAll() whatever it is? Since it mostly exists, it will do no harm right?

Reply:
BTW, this organization is what I was hoping for, thanks.

Reply:
Yes, we need to do all this logic because it's done *outside* the container, before `docker-compose build`.

And yes, we need to check that `working_dir` is set to some path inside `/var/www/html/`, otherwise we won't have a corresponding directory on the host side.

Reply:
I would be more comfortable with this if it were encapsulated a bit more. How about creating an app.GetHostWorkingDir(servicename) that does what needs to be done, removing the proper prefix. Then this code here would just be
```
d := GetHostWorkingDir("web")
os.MkDirAll(d, 0755)
```

I think it would be easier to follow.  

Another option is to return two values (in-container and on-host) from app.GetWorkingDir(). It's only used 5 places, so might not be too painful.